src/crypto/crypto.js

```diff
@@ -67,7 +75,24 @@ export async function publicKeyEncrypt(algo, publicParams, data, fingerprint) {
       const { oid, Q, kdfParams } = publicParams;
       const { publicKey: V, wrappedKey: C } = await publicKey.elliptic.ecdh.encrypt(
         oid, kdfParams, data, Q, fingerprint);
-      return { V, C: new ECDHSymkey(C) };
+      return { V, C: new ByteArray(C) };
+    }
+    case enums.publicKey.aead: {
+      if (!privateParams) {
+        throw new Error('Cannot encrypt with symmetric key missing private parameters');
+      }
+      const { cipher: algo } = publicParams;
+      const algoName = algo.getName();
+      const { keyMaterial } = privateParams;
+      const aeadMode = enums.read(enums.aead, config.aeadMode);
+      const mode = crypto[aeadMode];
+      const { tagLength, ivLength } = mode;
+      const iv = await getRandomBytes(ivLength);
+      const modeInstance = await mode(algoName, keyMaterial);
+      const ciphertext = await modeInstance.encrypt(data, iv, new Uint8Array());
+      const c = ciphertext.subarray(0, -tagLength);
+      const t = ciphertext.subarray(-tagLength);
+      return { aeadMode: new AEADEnum(aeadMode), iv, c: new ByteArray(c), t };
```

Comment:
I wouldn't encode the tag separately (i.e. I'd just keep it included in the ciphertext, the only thing it changes is the encoded length before, and I think it makes things simpler)

Reply:
The spec lists ciphertext and tag separately for AEAD packets (tag 20), I was going off that when writing out the draft RFC and this lines up with the draft. The symmetric encryption draft went from

<ul> <li>A symmetric key encryption of "m" dependent on cipher and AEAD mode</li>
                <li>An authentication tag generated by the AEAD mode."</li></ul>
to:
<ul> <li>A symmetric key encryption of "m" dependent on cipher and AEAD mode prefixed with a one-octet length </li>
                <li>An authentication tag generated by the AEAD mode."</li></ul>

It would now have to be something like:

<ul> <li>A symmetric key encryption of "m" dependent on cipher and AEAD mode concatenated with the authentication tag generated by the AEAD mode,  prefixed with a one-octet length of the concatenation</li>
              </ul>
It's a bit wordy :D.

Reply:
I think having the two separate is conceptually easier to understand and it aligns with the existing stuff in the spec.

Reply:
Are you referring to
```
   o  Encrypted data, the output of the selected symmetric-key cipher
      operating in the given AEAD mode.

   o  A final, summary authentication tag for the AEAD mode.
```
? Because these are two separate things. The first bullet point includes both the ciphertext and authentication tag of each chunk; the second bullet point is again a full invocation of the AEAD construct but on an empty plaintext, which means you only get an authentication tag.

Then in the text below it does mention the ciphertext and authentication tag separately, but not in separate bullet points, and the spec currently never asks you to split them up in code (they are always stored immediately next to each other and not counted separately). So I think we shouldn't do so here, either.